doc/coding-style-guide.md

```diff
@@ -318,11 +318,24 @@ func (controller *Controller) reconcileEdgeConnectDeletion(ctx context.Context,
 ### Do's
 
 - Write unit-tests ;)
+- Name test functions like the following
+  - for public funcs -> `func TestPublicFunc(t *testing.T)`
+  - for private funcs -> `func Test_privateFunc(t *testing.T)`
+  - for public method for public struct -> `func Test_MyStruct_PublicMethod(t *testing.T)`
+  - for private method for public struct -> `func Test_MyStruct_privateMethod(t *testing.T)`
+  - for public method for private struct -> `func Test_myStruct_PublicMethod(t *testing.T)`
+  - for private method for private struct -> `func Test_myStruct_privateMethod(t *testing.T)`
+  - Avoid "scenarios" in the name, use `t.Run` within the tests.
+    - So avoid funcs like `func Test_MyStruct_PublicMethod_Scenario1(t *testing.T)` and instead use `t.Run("scenario1", ...)` within `func Test_MyStruct_PublicMethod`
 - Use `assert` and `require`. (`github.com/stretchr/testify/assert, github.com/stretchr/testify/require`)
 - `require` is your friend, use it for checking errors (`require.NoError(t, err)`) or anywhere where executing the rest of the `assert`s in case of the check failing would just be noise in the output.
-- Abstract the setup/assert phase as much as possible so it can be reused in other tests in the package.
+- Mock per test, not per package.
+  - Having one `createTestReconciler(p1,p2,p3,p4,p5)` that tries to mock all and fit in all test cases is just maintenance hell, because if you change 1 test-case then you will change all others, also your mocks will be too forgiving, which is great for bugs.
+    - Create small, test specific mocks.
+    - Such mocks can be created with helper functions defined within the test case, to reuse it within the test case.
```

Comment:
I don't see it mentioned anywhere, but usage of t.Helper() is recommended for helpers to improve error reporting. The usage of it is currently a bit inconsistent.